For your reference:
```
- Performing Test LIBOMPTARGET_NVPTX_CUDA_COMPILER_SUPPORTS_FLAGS_REQUIRED
-- Performing Test LIBOMPTARGET_NVPTX_CUDA_COMPILER_SUPPORTS_FLAGS_REQUIRED - Failed
CMake Error at libomptarget/cmake/Modules/LibomptargetUtils.cmake:26 (message):
  LIBOMPTARGET: Cannot build CUDA LLVM bitcode offloading device RTL!
Call Stack (most recent call first):
  libomptarget/deviceRTLs/nvptx/CMakeLists.txt:126 (libomptarget_error_say)
```

So it does not get passed the second CMake invocation that builds the byte code. Here is the error that I see:
```
/scratch/naromero/opt/spack/linux-ubuntu18.04-westmere/gcc-7.5.0/llvm-11.0.0-p5vk7snzs2ynsjzaj7cyki4qjnkoszn6/bin/clang  -Wall -Wcast-qual -Wformat-pedantic -Wimplic\
it-fallthrough -Wsign-compare -Wno-extra -Wno-pedantic -DCHECK_FUNCTION_EXISTS=__atomic_load_1 -rdynamic CMakeFiles/cmTC_99f89.dir/CheckFunctionExists.c.o -o cmTC_99\
f89
ld: CMakeFiles/cmTC_99f89.dir/CheckFunctionExists.c.o: in function `main':
CheckFunctionExists.c:(.text+0x17): undefined reference to `__atomic_load_1'
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
CMakeFiles/cmTC_99f89.dir/build.make:105: recipe for target 'cmTC_99f89' failed
make[1]: *** [cmTC_99f89] Error 1

```

--- a/var/spack/repos/builtin/packages/llvm/package.py
+++ b/var/spack/repos/builtin/packages/llvm/package.py
@@ -198,7 +198,7 @@ class Llvm(CMakePackage, CudaPackage):
     # see  https://bugs.llvm.org/show_bug.cgi?id=39696
     # for a bug report about this problem in llvm master.
     patch("constexpr_longdouble.patch", when="@6:8+libcxx")
-    patch("constexpr_longdouble_9.0.patch", when="@9:+libcxx")
+    patch("constexpr_longdouble_9.0.patch", when="@9:10.0.0+libcxx")
 
     # Backport from llvm master; see
     # https://bugs.llvm.org/show_bug.cgi?id=38233

